### PR TITLE
JAMES-2930 Enrich and reorganize README for development

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -555,7 +561,7 @@ Where:
 
 Note: A helper script is provided for the generation of packages for a specific git commit. For instance:
 
-    $ sh dockerfiles/packaging/guice/cassandra/package.sh 3.2.0 1 c298195e84 $PWD/result
+    $ sh dockerfiles/packaging/guice/cassandra/package.sh 3.5.0 1 c298195e84 $PWD/result
```
Comment (rouazana) [inline README.adoc]: unneeded change, I guess there is a few chance that c298195e84 is matching 3.5.0 version.

File: README.adoc [hunk 2]
```diff
@@ -44,7 +44,7 @@ There is many other ways one can help us: packaging, communication, etc ...
  * https://github.com/apache/james-project/#how-to-release-via-maven-release-plugin[How to release via maven release plugin]
  * https://github.com/apache/james-project/#how-to-check-the-compilation[How to check the compilation]
  * https://github.com/apache/james-project/#how-to-run-james-in-docker[How to run James in Docker]
- ** https://github.com/apache/james-project/#run-james-with-java-8--guice--cassandra-rabbitmq-swift-elasticsearch[Run James with Java 8 + Guice + Cassandra + RabbitMQ + Swift + ElasticSearch]
+ ** https://github.com/apache/james-project/#run-james-with-java-8--guice--cassandra--rabbitmq--swift--elasticsearch[Run James with Java 8 + Guice + Cassandra + RabbitMQ + Swift + ElasticSearch]
```
Comment (rouazana) [inline README.adoc]: can you try with ++ (see https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#links )

File: README.adoc [hunk 3]
```diff
@@ -44,10 +44,10 @@ There is many other ways one can help us: packaging, communication, etc ...
...
- ** https://github.com/apache/james-project/#run-james-with-java-8--guice--cassandra--elasticsearch[Run James with Java 8 + Guice + Cassandra + ElasticSearch]
- ** https://github.com/apache/james-project/#run-james-with-java-8--guice--jpa--lucene[Run James with Java 8 + Guice + JPA + Lucene]
- ** https://github.com/apache/james-project/#run-james-with-java-8--spring--jpa[Run James with Java 8 + Spring + JPA]
+ ** ++https://github.com/apache/james-project/#run-james-with-java-8--guice--cassandra--rabbitmq--swift--elasticsearch++[Run James with Java 8 + Guice + Cassandra + RabbitMQ + Swift + ElasticSearch]
+ ** ++https://github.com/apache/james-project/#run-james-with-java-8--guice--cassandra--elasticsearch++[Run James with Java 8 + Guice + Cassandra + ElasticSearch]
+ ** ++https://github.com/apache/james-project/#run-james-with-java-8--guice--jpa--lucene++[Run James with Java 8 + Guice + JPA + Lucene]
+ ** ++https://github.com/apache/james-project/#run-james-with-java-8--spring--jpa[Run James with Java 8 + Spring + JPA]++
```
Comment (rouazana) [inline README.adoc]: that's worst, it's breaking formatting :/ please remove this last fixup, we'll fix this unrelated comment in another pr/issue

**Comment by Arsnael** [comment]
```
[9e54c95c656edd9ad2235c218f6c40e95d507014] [ERROR] Errors: 
[9e54c95c656edd9ad2235c218f6c40e95d507014] [ERROR]   SmtpRandomStoringTest.oneHundredMailsShouldHaveBeenStoredBetweenFourAndEightTimes:168 » ConditionTimeout
```
test this please

**Comment by blackheaven** [comment]
Maybe we can stop reaching a :green_apple: build for documentation?

**Comment by Arsnael** [comment]
> Maybe we can stop reaching a green_apple build for documentation?

Well it is :green_apple: now anyway :trollface:

File: README.adoc (hunk 2 above)
Comment (remk) [inline README.adoc]: This link still seems broken.

Comment (blackheaven) [reply README.adoc]: Yes, they will be fixed in another PR

**Pull request closed without merging**